apm-agent-core/src/main/java/co/elastic/apm/agent/logging/Log4jLoggerFactoryBridge.java

```diff
@@ -53,7 +56,12 @@ protected Logger newLogger(final String name, final LoggerContext context) {
 
     @Override
     protected LoggerContext getContext() {
-        final Class<?> anchor = StackLocatorUtil.getCallerClass(FQCN, PACKAGE);
-        return anchor == null ? LogManager.getContext() : getContext(StackLocatorUtil.getCallerClass(anchor));
+        // the logger context is defined by the class that calls LoggerFactory
+        final Class<?> factoryCaller = StackLocatorUtil.getCallerClass(LOGGER_FACTORY);
+        if (factoryCaller == null) {
+            return LogManager.getContext();
+        }
+        return getContext(factoryCaller);
```

Comment:
@eyalkoren I do need your review on this as I'm not 100% sure it's the right way to fix it.
Without this, the added anonymous class makes `loggingLevelChangeTest` test fail.